File: springboot_mp/src/main/java/com/xeon/springboot_mp/handler/FillMetaObjectHandler.java
package com.xeon.springboot_mp.handler;

import com.baomidou.mybatisplus.core.handlers.MetaObjectHandler;
import lombok.extern.slf4j.Slf4j;
import org.apache.ibatis.reflection.MetaObject;
import org.springframework.stereotype.Component;

import java.time.LocalDateTime;

@Slf4j
@Component
public class FillMetaObjectHandler implements MetaObjectHandler {


    /**
     * 数据插入时执行
     *
     * @param metaObject
     */
    @Override
    public void insertFill(MetaObject metaObject) {

        log.info("start insert fill ....");
        this.strictInsertFill(metaObject, "createTime", LocalDateTime::now, LocalDateTime.class); // 起始版本 3.3.3(推荐)
        this.strictInsertFill(metaObject, "updateTime", LocalDateTime::now, LocalDateTime.class); // 起始版本 3.3.3(推荐)
    }

    /**
     * 数据更新时执行
     *
     * @param metaObject
     */
    @Override
    public void updateFill(MetaObject metaObject) {
        log.info("start update fill ....");
        /*正常写法*/
        this.strictInsertFill(metaObject, "updateTime", () -> LocalDateTime.now(), LocalDateTime.class); // 起始版本 3.3.3(推荐)
    }
}

File: kotlin-springboot-kts/src/main/kotlin/com/yzq/kotlinspringbootkts/handler/FillMetaObjectHandler.kt
package com.yzq.kotlin_springboot_maven.handler

import com.baomidou.mybatisplus.core.handlers.MetaObjectHandler
import org.apache.ibatis.reflection.MetaObject
import org.springframework.stereotype.Component
import java.time.LocalDateTime


/**
 * @description: 默认数据填充
 * @author : yuzhiqiang (zhiqiang.yu.xeon@gmail.com)
 */

@Component
class FillMetaObjectHandler : MetaObjectHandler {
    override fun insertFill(metaObject: MetaObject) {
        this.strictInsertFill(
            metaObject, "createTime",
            { LocalDateTime.now() },//  ()->LocalDateTime.now()  或  LocalDateTime::now
            LocalDateTime::class.java
        )

        this.strictInsertFill(
            metaObject, "updateTime",
            { LocalDateTime.now() },
            LocalDateTime::class.java
        )

    }

    override fun updateFill(metaObject: MetaObject) {
        /*正常写法*/
        strictInsertFill(
            metaObject, "updateTime",
            { LocalDateTime.now() },
            LocalDateTime::class.java
        )

    }


}
